ui/tests/unit/routes/vault/cluster/oidc-callback-test.js

```diff
@@ -48,6 +49,108 @@ module('Unit | Route | vault/cluster/oidc-callback', function (hooks) {
     this.callbackUrlQueryParams('');
   });
 
+  module('getParamsForCallback helper fn', function () {
+    test('it parses params correctly with regular inputs no namespace', function (assert) {
+      const params = {
+        state: 'my-state',
+        code: 'my-code',
+        path: 'oidc-path',
+      };
+      const results = getParamsForCallback(params, '?code=my-code&state=my-state');
+      assert.deepEqual(results, { source: 'oidc-callback', ...params });
+    });
+
+    test('it parses params correctly regular inputs and namespace param', function (assert) {
+      const params = {
+        state: 'my-state',
+        code: 'my-code',
+        path: 'oidc-path',
+        namespace: 'my-namespace/nested',
+      };
+      const results = getParamsForCallback(params, '?code=my-code&state=my-state');
+      assert.deepEqual(results, { source: 'oidc-callback', ...params });
+    });
+
+    test('it parses params correctly regular inputs and namespace in state', function (assert) {
+      const queryString = '?code=my-code&state=my-state,ns=blah';
+      const params = {
+        state: 'my-state,ns', // mock what Ember does with the state QP
+        code: 'my-code',
+        path: 'oidc-path',
+      };
+      const results = getParamsForCallback(params, queryString);
+      assert.deepEqual(results, { source: 'oidc-callback', ...params, state: 'my-state', namespace: 'blah' });
+    });
+
+    test('namespace in state takes precedence over namespace in route', function (assert) {
+      const queryString = '?code=my-code&state=my-state,ns=some-ns/foo';
+      const params = {
+        state: 'my-state,ns', // mock what Ember does with the state QP
+        code: 'my-code',
+        path: 'oidc-path',
+        namespace: 'some-ns',
+      };
+      const results = getParamsForCallback(params, queryString);
+      assert.deepEqual(results, {
+        source: 'oidc-callback',
+        ...params,
+        state: 'my-state',
+        namespace: 'some-ns/foo',
+      });
+    });
+
+    test('correctly decodes path and state', function (assert) {
+      const searchString = '?code=my-code&state=spaces%20state,ns=some.ns%2Fchild';
+      const params = {
+        state: 'spaces state,ns',
+        code: 'my-code',
+        path: 'oidc-path',
+      };
+      const results = getParamsForCallback(params, searchString);
+      assert.deepEqual(results, {
+        source: 'oidc-callback',
+        state: 'spaces state',
+        code: 'my-code',
+        path: 'oidc-path',
+        namespace: 'some.ns/child',
+      });
+    });
+
+    test('correctly decodes namespace in state', function (assert) {
+      const queryString = '?code=my-code&state=spaces%20state,ns=spaces%20ns';
+      const params = {
+        state: 'spaces state,ns', // mock what Ember does with the state QP
```

Comment:
in this case it would be 
```suggestion
        state: 'spaces state,ns=spaces ns', // mock what Ember does with the state QP
```